### PR TITLE
aria2: 1.23.0 -> 1.24.0

### PR DESCRIPTION
**Pull request by womfoo**
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [x] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by romildo** [comment]
`autoreconfHook` can be in `nativeBuildInputs`.

**Comment by womfoo** [comment]
thanks @romildo 

* rebased changes to move `autoreconfHook` to `nativeBuildInputs`
* also added an upstream patch for fixing the OSX build

**Comment by joachifm** [comment]
@womfoo if possible, fetch the patch instead of adding it to the repository.

**Comment by womfoo** [comment]
@joachifm latest rebase now uses fetchpatch

File: pkgs/tools/networking/aria2/default.nix [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+    sha256 = "0sb8s2rf2l0x7m8fx8kys7vad0lfw3k9071iai03kxplkdvg96n9";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkgconfig autoreconfHook ];
 
   buildInputs = [ openssl c-ares libxml2 sqlite zlib libssh2 ] ++
     stdenv.lib.optional stdenv.isDarwin Security;
 
+  patches = [
```
Comment (joachifm) [inline pkgs/tools/networking/aria2/default.nix]: If this is to fix darwin, please make it conditional.

**Comment by womfoo** [comment]
fixed hash and made the patch osx-only

both build fine on linux and osx locally

```
[kranium@silverspark:~]$ uname -a
Linux silverspark 4.4.12 #1-NixOS SMP Wed Jun 1 19:16:06 UTC 2016 x86_64 GNU/Linux

[kranium@silverspark:~/git/github.com/womfoo/nixpkgs]$ nox-review pr 16362
=== Reviewing PR 16362 : aria2: 1.23.0 -> 1.24.0
==> We're in a git repo, trying to fetch it
==> Fetching base (master)
==> Fetching PR
==> Fetching extra history for merging
==> Merging PR into base
Building in /tmp/nox-review-48jcr3a8: uget aria
/nix/store/pfbj5zfkns9di6v4b31gbsllf6sv6wx3-uget-2.0.8
/nix/store/iy4rm9avv1cvh2fc6b90fkv9hyvg9lsk-aria2-1.24.0
Result in /tmp/nox-review-48jcr3a8
total 0
lrwxrwxrwx 1 kranium users 54 Jun 21 09:56 result -> /nix/store/pfbj5zfkns9di6v4b31gbsllf6sv6wx3-uget-2.0.8
lrwxrwxrwx 1 kranium users 56 Jun 21 09:56 result-2 -> /nix/store/iy4rm9avv1cvh2fc6b90fkv9hyvg9lsk-aria2-1.24.0
```

```
Kraniums-MacBook-Pro:nixpkgs kranium$ uname -a
Darwin Kraniums-MacBook-Pro.local 14.5.0 Darwin Kernel Version 14.5.0: Tue Sep  1 21:23:09 PDT 2015; root:xnu-2782.50.1~1/RELEASE_X86_64 x86_64
Kraniums-MacBook-Pro:nixpkgs kranium$ nox-review pr 16362
=== Reviewing PR 16362 : aria2: 1.23.0 -> 1.24.0
==> Creating nixpkgs repo in /Users/kranium/.nox/nixpkgs
==> We're in a git repo, trying to fetch it
==> Fetching base (master)
==> Fetching PR
==> Fetching extra history for merging
==> Merging PR into base
Building in /var/folders/lk/w3m5lzw94wj0hhtf9kxn45n40000gn/T/nox-review-3irqc935: aria
/nix/store/7gjvnzxmqw7nw76rh5sdsa8x903w1b0z-aria2-1.24.0
Result in /var/folders/lk/w3m5lzw94wj0hhtf9kxn45n40000gn/T/nox-review-3irqc935
total 8
lrwxr-xr-x  1 kranium  staff  56 Jun 21 10:46 result -> /nix/store/7gjvnzxmqw7nw76rh5sdsa8x903w1b0z-aria2-1.24.0
```

**Comment by joachifm** [comment]
Thank you

**Pull request merged**